tests/test_cli.py

```diff
@@ -262,6 +262,37 @@ def test_cli_secret_write_base64_ref(self):
 
         os.remove(test_tag_file)
 
+    def test_cli_secret_write_vault(self):
```

Comment:
Add test cases to cover other workflows/scenarios discussed in the previous comments